parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -330,6 +382,17 @@ public void write(int repetitionLevel, CharSequence value) {
     }
   }
 
+  private static class UUIDWriter extends PrimitiveWriter<UUID> {
+    private UUIDWriter(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public void write(int repetitionLevel, UUID value) {
+      column.writeBinary(repetitionLevel, Binary.fromReusedByteArray(UUIDUtil.convert(value)));
```

Comment:
The byte array passed here is allocated, not reused. It probably should be reused, like the one in Spark's writers.